command.go

```diff
@@ -685,11 +685,22 @@ func (cmd *ZSliceCmd) readReply(cn *pool.Conn) error {
 
 //------------------------------------------------------------------------------
 
+// Scanner is a ScanCmd with Iterator capabilities
+type Scanner struct {
+	*ScanCmd
+	c *commandable
+}
```

Comment:
That seems clumsy to me, that' why I added a new type wrapping ScanCmd. When client is the origin of your command, then you have an `Iterator` method, but you can also construct your own `ScanCmd` with `NewScanCmd` if you really want to. For me, the additional type is worth the flexibility. I am happy to rename `Scanner` to `ScanCmdIterable` or something more verbose.